test/simulator/main/org/apache/cassandra/simulator/systems/SimulatedTime.java

```diff
@@ -274,7 +274,7 @@ public long nanoTime()
                 nextDrift = nanosDriftSupplier.get(random);
                 from = global;
                 to = global + Math.max(baseDrift, nextDrift);
-                diffPerGlobal = (nextDrift - baseDrift) / (double)(to - from);
+                diffPerGlobal = to == from ? 1 : (nextDrift - baseDrift) / (double)(to - from);
```

Comment:
when i monkeyed with controlling time i hit this case... i wanted to test the case where the full test ran unrealistically fast (each operation only took 1ns)...

Reply:
I spent 5 minutes looking at this and didn't grok what `diffPerGlobal` is doing.

Reply:
Probably we should simulate non-monotonic system time changes